tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/internal/provisional/tmf/core/model/events/TmfEventTableDataProvider.java

```diff
@@ -152,15 +152,15 @@ public TmfModelResponse<TmfTreeModel<TmfEventTableColumnDataModel>> fetchTree(Ma
         for (ITmfEventAspect<?> aspect : aspects.values()) {
             synchronized (fAspectToIdMap) {
                 long id = fAspectToIdMap.computeIfAbsent(aspect, a -> createColumnId());
-                model.add(new TmfEventTableColumnDataModel(id, -1, Collections.singletonList(aspect.getName()), aspect.getHelpText(), aspect.isHiddenByDefault()));
+                model.add(new TmfEventTableColumnDataModel(id, -1, Collections.singletonList(aspect.getName()), aspect.getHelpText(), aspect.isHiddenByDefault(), aspect.getDataType().name()));
```

Comment:
This comment is for line 149 above where a `MultiAspect` can be created. A `MultiAspect` is created for aspects that are coming from the different traces and provide the same information, for example cpu, pid or tid. This avoids to have duplicated and redundant columns. The way how a `MuiltAspect` is created is that aspects have to have a common parent class. For example `LinuxTidAspect` is the base class for `KernelTidAspect` and  `ContextTidAspect`.

So, if one has an experiment with a LTTng UST and Kernel trace a `MultiAspect` is created and  currently it doesn't override the getDataType() method, and hence the data type that the `TmfEventTableDataProvider` provides is the default `STRING`.

Please update the `MultiAspect` to implement the `getDataType()` method and return the correct data type from the stored `fAspects` class variable (use the first one in the list). 
```java
@Override
    public DataType getDataType() {
        for (ITmfEventAspect<?> aspect : fAspects) {
            return aspect.getDataType();
        }
        return ITmfEventAspect.super.getDataType();
    }
```

Please also add a check in the `MultiAspect.create(aspects, aspectClass)` method if the list of aspects don't have all the same data type (which they should) similar to the warning if the names of aspects are not the same. See here what I mean https://github.com/eclipse-tracecompass/org.eclipse.tracecompass/blob/c8509f55bb32a5a7ad3ee7eb4ffe7d7828ac7316/tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/event/aspect/MultiAspect.java#L80).